pubsublite/spark-connector/README.md

```diff
@@ -0,0 +1,184 @@
+   > the virtualenv and go back to your global Python environment
+   > by running `deactivate`.
+
+1. Install the required packages.
+    ```bash
+    python -m pip install -U -r requirements.txt
+    ```
+
+## Creating a Spark cluster on Dataproc
+
+1. Go to [Cloud Console for Dataproc].
+
+1. Go to Clusters, then [Create Cluster].
+   > **Note:** When setting up the cluster, you must choose
+   > [Dataproc Image Version 1.5] under ___Versioning___ because
+   > the connector currently only supports Spark 2.4.8.
+   > Additionally, in ___Manage security (optional)___, you
+   > must enable the cloud-platform scope for your cluster by
+   > checking "Allow API access to all Google Cloud services in
+   > the same project" under ___Project access___.
+
+   Here is an equivalent example using a `gcloud` command, with an additional optional argument to enable component gateway:
+
+    ```sh
+    export CLUSTER_ID=your-cluster-id
+    export DATAPROC_REGION=your-dataproc-region
+
+    gcloud dataproc clusters create $CLUSTER_ID \
+      --region $DATAPROC_REGION \
+      --image-version 1.5-debian10 \
+      --scopes 'https://www.googleapis.com/auth/cloud-platform' \
+      --enable-component-gateway
+    ```
+
+## Writing to Pub/Sub Lite
+
+[spark_streaming_to_pubsublite_example.py](spark_streaming_to_pubsublite_example.py) creates a streaming source of consecutive numbers with timestamps for 60 seconds and writes them to a Pub/Sub topic.
+
+To submit a write job:
+
+```sh
+export PROJECT_NUMBER=$(gcloud projects list --filter="projectId:$PROJECT_ID" --format="value(PROJECT_NUMBER)")
+
+gcloud dataproc jobs submit pyspark spark_streaming_to_pubsublite_example.py \
+    --region=$DATAPROC_REGION \
+    --cluster=$CLUSTER_ID \
+    --jars=gs://spark-lib/pubsublite/pubsublite-spark-sql-streaming-LATEST-with-dependencies.jar \
+    --driver-log-levels=root=INFO \
+    --properties=spark.master=yarn \
+    -- $PROJECT_NUMBER $PUBSUBLITE_LOCATION $TOPIC_ID
+```
+
+Visit the job URL in the command output or the jobs panel in [Cloud Console for Dataproc] to monitor the job progress.
+
+You should see INFO logging like the following in the output:
+
+```none
+INFO com.google.cloud.pubsublite.spark.PslStreamWriter: Committed 1 messages for epochId ..
+```
+
+## Reading from Pub/Sub Lite
+
+[spark_streaming_from_pubsublite_example.py](spark_streaming_from_pubsublite_example.py) reads messages formatted as dataframe rows from a Pub/Sub subscription and prints them out to the console.
+
+To submit a read job:
+
+```sh
+gcloud dataproc jobs submit pyspark spark_streaming_from_pubsublite_example.py \
+    --region=$DATAPROC_REGION \
+    --cluster=$CLUSTER_ID \
+    --jars=gs://spark-lib/pubsublite/pubsublite-spark-sql-streaming-LATEST-with-dependencies.jar \
+    --driver-log-levels=root=INFO \
+    --properties=spark.master=yarn \
+    -- $PROJECT_NUMBER $PUBSUBLITE_LOCATION $SUBSCRIPTION_ID
+```
+
+Here is an example output:
+
+```none
```

Comment:
add a TODO here to modify attributes once that's fixed.

Reply:
Added a TODO.